### PR TITLE
feat(shorebird_cli): add ability to release with `--no-codesign`

### LINKED ISSUE #936: fix: IOS build for release with error: "Runner.app" requires a provisioning profile with the Associated Domains feature.
**Description**

run "shorebird release ios-alpha", It will toast an error
"Runner.app" requires a provisioning profile with the Associated Domains feature.

**Steps To Reproduce**

1. shorebird release ios-alpha
2. watch board

**Expected Behavior**

build successfully

**Screenshots**

image

**Additional Context**

in xcode, it will build successfully
image

### LINKED ISSUE #1067: feat: `shorebird release ios-alpha` fails when 'Automatically manage signing' off
**Description**

##### Steps to Reproduce

1. In an project configured to use Shorebird, turn off `Automatically manage signing` for the `Runner` target's `Signing & Capabilities` in `Runner.xcodeproj`.
2. Run `shorebird release ios-alpha`. 

Expected result: The release is created.

Actual result: The release cannot be created.

<details><summary>Output</summary>
<p>

```
[WARN] iOS support is alpha. See https://docs.shorebird.dev/status#ios-alpha for more information.
[HTTP] GET https://api.shorebird.dev/api/v1/apps
✓ Fetching apps (0.2s)
[Process.run] /Users/[redacted]/.shorebird/bin/cache/flutter/0e2d280277cf9f60f7ba802a59f9fd187ffdd050/bin/flutter build ipa --release --export-options-plist=/var/folders/r8/txtblszs2b7c67rhw8fbj8zm0000gq/T/X7Trep/ExportOptions.plist
✗ Failed to build IPA (100.1s)
    "Runner.app" requires a provisioning profile with the Associated Domains and Push Notifications features.
```

</p>
</details> 

##### Root Cause

The `ExportOptions.plist` file contains this key-value pair which is incompatible with the pbxproj project configuration for Runner of `CODE_SIGN_STYLE = Manual`.

```xml
  <key>signingStyle</key>
  <string>automatic</string>
```

**Requirements**

- [ ] Emit a proper value for the `signingStyle` entry in the generated `ExportOptions.plist` by inspecting the value of `CODE_SIGN_STYLE` for the Runner target

There may be other approaches, such as omitting the entry altogether as `flutter build ios` [does by default](https://github.com/flutter/flutter/blob/e2badf38570db7ee6526b778549d300b7d1c9ba5/packages/flutter_tools/lib/src/commands/build_ios.dart#L536).

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

- Adds the `--no-codesign` flag to `shorebird release ios-alpha` to allow users to manage their own codesigning.
- Removes ipa files as a source of truth, uses xcarchives instead.

Fixes https://github.com/shorebirdtech/shorebird/issues/1067
Fixes https://github.com/shorebirdtech/shorebird/issues/936

## Known Limitations

- Because the binary is not codesigned, `shorebird preview` will not work
- If release and patch commands are not provided the same value for the `codesign` flag, the patch command will erroneously report native code changes.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/release/release_ios_command.dart [hunk 1]
```diff
@@ -193,9 +200,18 @@ ${summary.join('\n')}
       status: ReleaseStatus.active,
     );
 
-    logger
-      ..success('\n✅ Published Release!')
-      ..info('''
+    logger.success('\n✅ Published Release!');
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/release/release_ios_command.dart:212]: shouldn't we move the success log until after we've checked the ipaPath? Seems weird to potentially output "success" followed by failure.

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/release/release_ios_command.dart:212]: Good call, although I think the right solution is to move this up a bit so we don't publish a release if we failed to generate an ipa. Fixing now

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/release/release_ios_command.dart:212]: Fixed.

File: packages/shorebird_cli/lib/src/archive_analysis/ios_archive_differ.dart [hunk 2]
```diff
@@ -102,8 +104,7 @@ class IosArchiveDiffer extends ArchiveDiffer {
     }
 
     final outFile = File(outPath);
-    final hash = _hash(outFile.readAsBytesSync());
-    return hash;
+    return _hash(outFile.readAsBytesSync());
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/archive_analysis/ios_archive_differ.dart:107]: 💯

File: packages/shorebird_cli/lib/src/commands/release/release_ios_command.dart [hunk 3]
```diff
@@ -71,6 +72,17 @@ make smaller updates to your app.
...
     showiOSStatusWarning();
 
+    final codesign = results['codesign'] == true;
+    if (!codesign) {
+      logger
+        ..info(
+          '''Building for device with codesigning disabled. You will have to manually codesign before deploying to device.''',
+        )
+        ..warn(
+          '''shorebird preview does not work for releases created with codesigning disabled. However, you can still preview your app by signing the generated .xcarchive in Xcode.''',
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/release/release_ios_command.dart]:
```suggestion
          '''shorebird preview will not work for releases created with "--no-codesign". However, you can still preview your app by signing the generated .xcarchive in Xcode.''',
```

**Review by felangel — APPROVED** [review]

**Pull request merged**